factory/__main__.py

```diff
@@ -28,26 +30,51 @@ def main():
         add_gruvbox_palette(gruvbox_factory, palette)
 
     for image_path in image_paths:
-        if os.path.isfile(image_path):
-            process_image(image_path, console, gruvbox_factory)
-        else:
+        folder, file = split(image_path)
+        if file == "*":
+            subfolder = join(dirname(split(image_path)[0]), split(folder)[1])
+            # assert that image_path is a folder that can be iterated through
+            if isdir(subfolder):
+                for i in listdir(folder):
+                    # TODO: make a check to see that 'i' ends with png, jpg, etc ...
+                    process_image(join(subfolder, i), console, gruvbox_factory)
+                continue
+            else:
+                console.print(
+                    f"❌ [red]We had a problem in the pipeline! \n'{split(folder)[1]}' is not a folder that can be iterated through! \nSkipping... [/]"
+                )
+                continue
+        elif isdir(image_path):
             console.print(
-                f"❌ [red]We had a problem in the pipeline! \nThe image at '{image_path}' could not be found! \nSkipping... [/]"
+                    f"⏭️ [yellow]Skipping folder: {split(folder)[1]}... [/]"
```

Comment:
I still prefer the pipeline thing, so it's more in the "factory" lore hahahaha I think we can keep the ⏭️ emoji and yellow color, but use the previously